src/main/java/com/lorepo/icplayer/client/printable/PrintableContentParser.java

```diff
@@ -387,9 +388,20 @@ public String generatePrintableHTMLForPages(List<Page> sourcePages, boolean rand
 		result += "</div>";
 		return result;
 	};
-	
-	public String generatePrintableHTML(Content contentModel, boolean randomizePages, boolean randomizeModules, boolean showAnswers) {
-		List<Page> pages = contentModel.getPages().getAllPages();
+
+	public String generatePrintableHTML(Content contentModel, ArrayList<Integer> pageSubset, boolean randomizePages, boolean randomizeModules, boolean showAnswers) {

```

Comment:
Sugeruję: https://refactoring.guru/introduce-parameter-object + https://dzone.com/articles/design-patterns-the-builder-pattern
Ale się nie upieram;-)